signac/contrib/job.py

```diff
@@ -383,7 +380,18 @@ def reset_statepoint(self, new_statepoint):
             The job's new state point.
 
         """
-        self._statepoint.reset(new_statepoint)
+        if self._statepoint_requires_init:
+            # Instantiate state point data lazily - no load is required, since
+            # we are provided with the new state point data.
+            self._statepoint = _StatePointDict(
```

Comment:
Do we not have any tests that open a job by id (without a state point)? There's some surprising missing coverage here, was this one of the issues you noted on a previous PR? This seems new (most of the other ones were exception cases in `init`). I'm fine addressing in a separate PR to keep our optimization push moving, but if this is actually a mising case can you note it on #473?

Reply:
We definitely have tests for opening a job by id. The missing coverage is for "resetting a statepoint when opened by id, without accessing the state point or calling `job.init()`. I will add a test for that case.

Reply:
I added a test and fixed a bug. Thanks for the catch @vyasr 😄

Reply:
Thanks! Good fix.